### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by JassonCordones**
Improves error handling in some places, mainly in steam service. There was a bug when trying to reload games.
Minor typo fix.

### CODE REVIEW

File: lutris/util/jobs.py [hunk 1]
```diff
@@ -15,12 +15,12 @@ def __init__(self, func, callback, *args, **kwargs):
...
         self.stop_request = threading.Event()
-
-        super().__init__(target=self.target, args=args, kwargs=kwargs)
         self.function = func
         self.callback = callback if callback else lambda r, e: None
-        self.daemon = kwargs.pop("daemon", True)
 
+        super().__init__(target=self.target, args=args, kwargs=kwargs)
+        #self.daemon = kwargs.pop("daemon", True)
+        self.setDaemon(kwargs.pop("daemon", True))
```
Comment (danieljohnson2) [inline lutris/util/jobs.py:23]: I think this change is wrong; we support back to Python 3.7, and by then setDaemon had been deprecated. You are supposed to set the self.daemon property.

So why setDaemon?

Also, moving the call to ``super().__init__`` later seems harmless, but why?

File: bin/lutris [hunk 2]
```diff
@@ -47,18 +47,22 @@ try:
...
 
 if "WEBKIT_DISABLE_DMABUF_RENDERER" not in os.environ:
     os.environ["WEBKIT_DISABLE_DMABUF_RENDERER"] = "1"
 
-from lutris.gui.application import Application  # pylint: disable=no-name-in-module
+try:
+    from lutris.gui.application import Application  # pylint: disable=no-name-in-module
+except ImportError as ex:
+    sys.stderr.write("Error importing Lutris application module: %s\n" % ex)
+    sys.exit(1)
```
Comment (danieljohnson2) [inline bin/lutris:65]: Why do we want to catch this exception? We're going to exit if this fails in any case. What does this handler add?

File: lutris/services/steam.py [hunk 3]
```diff
@@ -127,29 +127,33 @@ def install_from_steam(self, manifest):
...
+                installed=1,
+                installer_slug=lutris_game_id,
+                configpath=configpath,
+                platform="Linux",
+                service=self.id,
+                service_id=appid,
+            )
+            return slug
+        except Exception as e:
+            logger.error("Failed to install from Steam: %s", str(e))
```
Comment (danieljohnson2) [inline lutris/services/steam.py:155]: I don't think we need ``str(ex)`` here- just ``ex`` should work. That's what we usually have for ``logger.error``.

**Review by danieljohnson2 — CHANGES_REQUESTED** [review]
I like most of this- it's mostly log-and-continue-to-the-next-game handling, which seems helpful. But I have a few concerns before I'll merge this.

The big one is switching us over to a deprecated method in jobs.py. I don't like that.

File: bin/lutris (hunk 2 above)
Comment (JassonCordones) [reply bin/lutris:65]: Here I was just trying to make the import error explicit, but it's true the program will exit anyways if you want I can revert it back.

File: lutris/util/jobs.py (hunk 1 above)
Comment (JassonCordones) [reply lutris/util/jobs.py:23]: I'll change it back sorry, I was not aware. 

For moving the call to `super().__init__` later , honestly I just liked how it looked better. I can revert it back too if you want.

File: lutris/services/steam.py (hunk 3 above)
Comment (JassonCordones) [reply lutris/services/steam.py:155]: I'll make the change

**Comment by danieljohnson2** [comment]
I will merge if we just go back to self.daemon instead of self.setDaemon; but I'd make those other changes too if it were my PR.

**Review by danieljohnson2 — APPROVED** [review]

**Comment by danieljohnson2** [comment]
Merged it. Thanks for your contribution!

**Pull request merged**